lib/elixir/lib/io.ex

```diff
@@ -301,37 +311,44 @@ defmodule IO do
 
   """
   @spec stream(device, :line | pos_integer) :: Enumerable.t
-  def stream(device, line_or_codepoints) do
+  def stream(device, line_or_codepoints)
+  when line_or_codepoints == :line
+  or is_integer(line_or_codepoints) and line_or_codepoints > 0 do
```

Comment:
Maybe this would look cleaner if we put parens around the `and` expression? Just to make precedence very clear.